bfvmm/src/serial/include/serial_port_x86.h

```diff
@@ -0,0 +1,210 @@
+//
+// Bareflank Hypervisor
+//
+// Copyright (C) 2015 Assured Information Security, Inc.
+// Author: Rian Quinn        <quinnr@ainfosec.com>
+// Author: Brendan Kerrigan  <kerriganb@ainfosec.com>
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+#ifndef SERIAL_X86__H
+#define SERIAL_X86__H
+
+#include <port_io.h>
+#include <serial_port.h>
+/**
+ *  @file    serial.h
+ *  @author  Brendan Kerrigan
+ *  @date    2015/11/02
+ *  @version 0.1
+ *
+ *  @brief Provide serial port access
+ *
+ *  @section DESCRIPTION
+ *
+ *  Provides a wrapper class around a serial port, which is
+ *  accessed via port I/O.
+ *
+ *  Useful information:
+ *  http://wiki.osdev.org/Serial_Ports
+ *  http://www.sci.muni.cz/docs/pc/serport.txt
+ *
+ */
+
+#define COM1_IO_PORT 0x3f8
+#define COM2_IO_PORT 0x2f8
+#define COM3_IO_PORT 0x3e8
+#define COM4_IO_PORT 0x2e8
+
+/* Limits */
+#define MIN_DATA_SIZE 5
+#define MAX_DATA_SIZE 8
+
+#define MIN_STOP_BITS 2
+#define MAX_STOP_BITS 2
+
+#define MIN_INTERRUPT 0
+#define MAX_INTERRUPT 15
+
+/* Baud rate defines and lookup tables */
+#define MIN_BAUD_RATE 50
+#define MAX_BAUD_RATE 115200
+
+struct __divisor_rate
+{
+    uint32_t baud_rate;
+    uint16_t divisor;
+};
+
+extern struct __divisor_rate divisor_table[];
+
+/* Parity defines and lookup tables */
+
+#define PARITY_MASK             0xC7
+
+struct __parity_bits
+{
+    PARITY_MODE mode;
+    uint8_t parity_bits;
+};
+
+extern struct __parity_bits parity_table[];
+
+/* Data size defines and lookup tables */
+#define DATA_SIZE_MASK          0xFC
+
+struct __data_bits
+{
+    uint8_t data_size;
+    uint8_t data_bits;
+};
+
+extern struct __data_bits data_size_table[];
+
+/* Receive buffer register */
```

Comment:
I guess I'm fine with the use of C comments for large blocks, but that doesn't explain it's use for single line comments, which is clearly more difficult. But yeah, my biggest complaint with old C style comments is you cannot block comment large portions of code with C style comments everywhere. With C++ comments, you can comment, comments just fine. (granted, you can use #if 0 macros, but I just find that more of a pain